### ISSUE TITLE
Working with custom sources without connecting to mapbox

### ISSUE THREAD

**Issue by pnorman** [mapbox/mapbox-studio-classic#583]
I'm looking at using TM2 with custom sources, without connecting to a Mapbox account. How can I do so?

**Comment by yhahn** [comment]
@pnorman this isn't possible atm, mostly for our own sanity as developers (ie. to avoid if/else branches evwhere in the code).

The Mapbox account is only used for uploading to Mapbox in the source editor which I'm assuming you won't need to do, so using a free account will be fine.

**Comment by pnorman** [comment]
Does this also mean that it's now impossible to use TM2 offline?

**Comment by yhahn** [comment]
No, once you have oauth credentials (which are stored locally) you can work offline.

**Comment by pnorman** [comment]
Is there documentation of what you need to do besides `npm install` to get TM2 working, e.g. pre-trip?

**Comment by yhahn** [comment]
Not atm, if you would like to add a HOWTO would be much appreciated https://github.com/mapbox/mapbox-studio

**Comment by mpmckenna8** [comment]
I tried to do a get up and running with tm2 post http://whywhathowdo.blogspot.com/2014/04/tilemill-2-quick-start-for-os-x.html but it's mainly focused at an osx homebrew crowd. I think a few things might have changed too.

**Comment by yhahn** [comment]
@mpmckenna8 install instructinos here are up to date:

https://github.com/mapbox/mapbox-studio/blob/master/README.md#install

**Comment by pnorman** [comment]
This issue is also focused on the plane trip scenario, where you need to download everything you will need in advance, because internet access is unavailable or extremely expensive.
